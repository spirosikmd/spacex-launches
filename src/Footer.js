import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import { Typography } from '@material-ui/core';
import Anchor from './Anchor';

const styles = theme => ({
  root: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column',
    padding: theme.spacing.unit * 2,
  },
});

const Footer = ({ classes }) => {
  return (
    <div className={classes.root}>
      <Typography>
        SpaceX Launches |{' '}
        <Anchor target="_blank" href="https://spyros.io">
          spyros.io
        </Anchor>
      </Typography>
      <Typography>
        Made possible by the{' '}
        <Anchor target="_blank" href="https://github.com/r-spacex/SpaceX-API">
          SpaceX API
        </Anchor>
      </Typography>
    </div>
  );
};

export default withStyles(styles)(React.memo(Footer));
