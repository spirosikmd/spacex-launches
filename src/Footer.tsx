import React from 'react';
import PropTypes from 'prop-types';
import {
  withStyles,
  Theme,
  WithStyles,
  createStyles,
} from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Anchor from './Anchor';

const styles = (theme: Theme) =>
  createStyles({
    root: {
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      flexDirection: 'column',
      padding: theme.spacing(2),
    },
  });

const Footer = ({ classes }: WithStyles<typeof styles>) => {
  return (
    <div className={classes.root}>
      <Typography>
        SpaceX Launches |{' '}
        <Anchor target="_blank" href="https://spyros.io">
          spyros.io
        </Anchor>
      </Typography>
      <Typography>
        Made possible by the{' '}
        <Anchor target="_blank" href="https://github.com/r-spacex/SpaceX-API">
          SpaceX API
        </Anchor>
      </Typography>
    </div>
  );
};

Footer.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(React.memo(Footer));
