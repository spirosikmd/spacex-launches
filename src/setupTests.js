import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from 'enzyme-to-json';
import { unwrap } from '@material-ui/core/test-utils';

configure({ adapter: new Adapter() });

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

global.shallow = (Component, props = {}) => {
  const ComponentWithoutStyles = unwrap(Component);
  return shallow(<ComponentWithoutStyles {...props} />);
};

global.mount = (Component, props = {}) => {
  const ComponentWithoutStyles = unwrap(Component);
  return mount(<ComponentWithoutStyles {...props} />);
};

global.Date = jest.fn().mockImplementation(() => {
  return {
    toUTCString: jest.fn().mockReturnValue('Fri, 11 Jan 2019 15:31:00 GMT'),
    getDate: jest.fn().mockReturnValue(3),
    getMonth: jest.fn().mockReturnValue(10),
    getFullYear: jest.fn().mockReturnValue(2019),
  };
});

// https://github.com/airbnb/enzyme/issues/1875
jest.mock('react', () => {
  const r = jest.requireActual('react');
  return { ...r, memo: x => x };
});
