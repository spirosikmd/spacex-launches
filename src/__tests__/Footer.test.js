import React from 'react';
import { render } from '../setupTests';
import Footer from '../Footer';

describe('Footer', () => {
  it('render a footer', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
