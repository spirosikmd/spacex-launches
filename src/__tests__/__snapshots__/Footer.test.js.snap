// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders 1`] = `
<div
  className="root"
>
  <WithStyles(Typography)>
    SpaceX Launches |
     
    <WithStyles(Component)
      href="https://spyros.io"
      target="_blank"
    >
      spyros.io
    </WithStyles(Component)>
  </WithStyles(Typography)>
  <WithStyles(Typography)>
    Made possible by the
     
    <WithStyles(Component)
      href="https://github.com/r-spacex/SpaceX-API"
      target="_blank"
    >
      SpaceX API
    </WithStyles(Component)>
  </WithStyles(Typography)>
</div>
`;
